thirdeye-server/src/main/java/ai/startree/thirdeye/auth/AuthorizationManager.java

```diff
@@ -58,6 +62,12 @@ public class AuthorizationManager {
   private final EnumerationItemManager enumerationItemManager;
   private final AnomalyManager anomalyManager;
 
+  // cache resource id's to their respective auth namespaces
+  private final Cache<Long, String> namespaceCache = CacheBuilder.newBuilder()
+      .maximumSize(2048)
```

Comment:
Given its long vs string. this should be a small map. How did we decide on size 2k?

Reply:
I just picked a number. Haven't settled on the _right_ number yet. Since the map should be small, would it make sense to cache namespaces for all entities up front?